src/pages/index.js

```diff
@@ -82,9 +46,195 @@ const addPlaceModal = new PopupWithForm(
   handleNewPlaceSubmit
 );
 
+const updateAvatarModal = new PopupWithForm(
+  "#avatar-update-modal",
+  handleAvatarSubmit
+);
+const imagePreviewModal = new PopupWithImage("#places-preview-modal");
+const deletePlaceModal = new PopupWithForm("#places-delete-modal", () => {});
+
 editProfileModal.setEventListeners();
 imagePreviewModal.setEventListeners();
+updateAvatarModal.setEventListeners();
 addPlaceModal.setEventListeners();
-section.renderItems();
+
+deletePlaceModal.setEventListeners();
+avatarValidation.enableValidation();
 profileEditValidation.enableValidation();
 addPlaceValidation.enableValidation();
+
+function handleSubmit(
+  request,
+  popupInstance,
+  loadingText = "Saving...",
+  defaultText = "Save"
+) {
+  popupInstance.setLoading(true, loadingText);
+
+  return request()
+    .then(() => {
+      popupInstance.close();
+    })
+    .catch((error) => {
+      console.error("Error:", error);
+    })
+    .finally(() => {
+      popupInstance.setLoading(false, defaultText);
+    });
+}
+
+function handleDeleteClick(cardID, cardElement) {
+  deletePlaceModal.setSubmitHandler(() => {
+    function makeRequest() {
+      return api.removePlace(cardID).then(() => {
+        cardElement.remove();
+      });
+    }
+
+    handleSubmit(makeRequest, deletePlaceModal, "Removing...", "Yes");
+  });
+  deletePlaceModal.open();
+}
+
+function getCardElement(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeReact
+  );
+  return card.getView();
+}
+
+/* GET Cards */
+let section;
+api
+  .getInitialCards()
+  .then((items) => {
+    section = new Section(
+      {
+        items: items,
+        renderer: (cardData) => {
+          const cardElement = getCardElement(cardData);
+          return cardElement;
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
+  })
+  .catch((err) => console.error(err));
+
+/* GET Profile */
+const userInfo = new UserInfo();
+api
+  .getUserInfo()
+  .then((profileData) => {
+    userInfo.setUserInfo(profileData);
+  })
+  .catch((err) => console.error(err));
+
+/* PATCH Profile Edit Function */
+function handleProfileEditSubmit(profileInputValues) {
+  editProfileModal.setLoading(true);
+  const userData = {
+    name: profileInputValues.title,
+    about: profileInputValues.description,
+  };
+
+  api
+    .editUserInfo(userData.name, userData.about)
+    .then((updatedUserData) => {
+      userInfo.setUserInfo(updatedUserData);
+      editProfileModal.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user info:", err);
+    })
+    .finally(() => {
+      editProfileModal.setLoading(false);
+    });
```

Comment:
**Could be improved**

You can use `handleSubmit` everywhere not to duplicate code